### PR TITLE
feat(python): add support for poetry dev dependencies

### PR DESCRIPTION
**Pull request by nikpivkin**
## Related PRs
- [x] https://github.com/aquasecurity/trivy/pull/8106

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/analyzer/language/python/poetry/poetry.go [hunk 1]
```diff
@@ -104,45 +104,46 @@ func (a poetryAnalyzer) mergePyProject(fsys fs.FS, dir string, app *types.Applic
...
 }
 
-func filterProdPackages(project pyproject.PyProject, app *types.Application) {
+func getDevDeps(project pyproject.PyProject) set.Set[string] {
+	deps := set.New[string]()
+	for _, groupDeps := range project.Tool.Poetry.Groups {
+		deps.Append(groupDeps.Dependencies.Items()...)
+	}
+	return deps
+}
```
Comment (DmitriyLewen) [inline pkg/fanal/analyzer/language/python/poetry/poetry.go:130]: I suggest 2 changes:
1. rename function (see https://go.dev/doc/effective_go#Getters)
2. move `project.Tool.Poetry.Dependencies` into function.
```suggestion
func directDeps(project pyproject.PyProject) set.Set[string] {
	deps := project.Tool.Poetry.Dependencies
	for _, groupDeps := range project.Tool.Poetry.Groups {
		deps.Append(groupDeps.Dependencies.Items()...)
	}
	return deps
}
```

File: pkg/fanal/analyzer/language/python/poetry/poetry.go [hunk 2]
```diff
@@ -104,45 +104,46 @@ func (a poetryAnalyzer) mergePyProject(fsys fs.FS, dir string, app *types.Applic
...
+
+func getProdPackages(app *types.Application, prodRootDeps set.Set[string]) set.Set[string] {
```
Comment (DmitriyLewen) [inline pkg/fanal/analyzer/language/python/poetry/poetry.go:132]:
```suggestion
func prodPackages(app *types.Application, prodRootDeps set.Set[string]) set.Set[string] {
```

File: docs/docs/coverage/language/python.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ The following table provides an outline of the features Trivy offers.
 |-----------------|------------------|:-----------------------:|:----------------:|:------------------------------------:|:--------:|:----------------------------------------:|
 | pip             | requirements.txt |            -            |     Include      |                  -                   |    ✓     |                    ✓                     |
 | Pipenv          | Pipfile.lock     |            ✓            |     Include      |                  -                   |    ✓     |                Not needed                |
-| Poetry          | poetry.lock      |            ✓            |     Exclude      |                  ✓                   |    -     |                Not needed                |
+| Poetry          | poetry.lock      |            ✓            |     Include      |                  ✓                   |    -     |                Not needed                |
```
Comment (DmitriyLewen) [inline docs/docs/coverage/language/python.md:29]: link for `Include` (https://github.com/aquasecurity/trivy/pull/8134#discussion_r1896422468)

Comment (nikpivkin) [reply docs/docs/coverage/language/python.md:29]: I've added a link for all package managers in [this](https://github.com/aquasecurity/trivy/pull/8134) PR

File: docs/docs/coverage/language/python.md [hunk 4]
```diff
@@ -26,7 +26,7 @@ The following table provides an outline of the features Trivy offers.
 |-----------------|------------------|:-----------------------:|:----------------:|:------------------------------------:|:--------:|:----------------------------------------:|
 | pip             | requirements.txt |            -            |     Include      |                  -                   |    ✓     |                    ✓                     |
 | Pipenv          | Pipfile.lock     |            ✓            |     Include      |                  -                   |    ✓     |                Not needed                |
-| Poetry          | poetry.lock      |            ✓            |     Exclude      |                  ✓                   |    -     |                Not needed                |
+| Poetry          | poetry.lock      |            ✓            |     [Include](#poetry)      |                  ✓                   |    -     |                Not needed                |
```
Comment (knqyf263) [inline docs/docs/coverage/language/python.md:29]: ditto
https://github.com/aquasecurity/trivy/pull/8134#discussion_r1896686174

Comment (nikpivkin) [reply docs/docs/coverage/language/python.md]: Fixed https://github.com/aquasecurity/trivy/pull/8152/commits/b5bafdab31fc752cec985035ecd181da351f21b0

**Review by knqyf263 — APPROVED** [review]
LGTM

**Pull request merged**